spyder/app/mainwindow.py

```diff
@@ -567,7 +569,24 @@ def create_toolbar(self, title, object_name, iconsize=24):
         self.toolbarslist.append(toolbar)
         return toolbar
 
-    def setup(self):
+    def setup_menus(self):

```

Comment:
@juanis2112, to avoid writing almost the same instructions several times in this method, you can rewrite it like this:

```python
menus_names = ['file', 'edit', 'search', 'source', 'run', 'debug', 'consoles',
               'projects', 'tools', 'help']
if sys.platform == 'darwin':
    for name in menu_names:
        menu = getattr(self, name + '_menu')
        menu_actions = getattr(self, name + '_menu_actions')
        menu.aboutToShow.connect(lambda menu_actions=menu_actions:
                                 self.verify_menu_actions(menu_actions, False))
```

`getattr` allows you to get an attribute of an object (in this case `self`) by passing a string to it. So it's very useful to manipulate attributes in a programmatic way (like in this case).

Reply:
Also, please move this method to be after the definition of the `setup` method and not before (as you added it here).

Reply:
Finally, please move these lines here too:

https://github.com/spyder-ide/spyder/blob/09970e054d7cfebcf7e7802b674d9b7c70d8db8c/spyder/app/mainwindow.py#L1206-L1216

to this method too.